### PR TITLE
Adds repo information to error message for fetch failure

### PR DESCRIPTION
**Pull request by iAmWillShepherd**
Context is in this comment: https://github.com/desktop/desktop/issues/5304#issuecomment-409535297

### CODE REVIEW

File: app/src/lib/stores/helpers/background-fetcher.ts [hunk 1]
```diff
@@ -96,7 +96,10 @@ export class BackgroundFetcher {
       try {
         await this.fetch(this.repository)
       } catch (e) {
-        log.error('Error performing periodic fetch', e)
+        log.error(
+          `Error performing periodic fetch for '${this.repository.name}'`,
```
Comment (billygriffin) [inline app/src/lib/stores/helpers/background-fetcher.ts]: For my edification does this expose that message to the user as well?

Comment (iAmWillShepherd) [reply app/src/lib/stores/helpers/background-fetcher.ts]: Nope, this will be written to logs.

Comment (shiftkey) [inline app/src/lib/stores/helpers/background-fetcher.ts]: Did we want to differentiate here between `owner/name` repositories and `folder` repositories, if `githubRepository` is defined?

File: app/src/lib/stores/helpers/background-fetcher.ts [hunk 2]
```diff
@@ -96,7 +96,13 @@ export class BackgroundFetcher {
       try {
         await this.fetch(this.repository)
       } catch (e) {
-        log.error('Error performing periodic fetch', e)
+        const ghRepo = this.repository.gitHubRepository
+        const repoName =
+          ghRepo !== null
+            ? `${ghRepo.owner.login}/${ghRepo.name}`
```
Comment (shiftkey) [inline app/src/lib/stores/helpers/background-fetcher.ts]: :art: This can be replaced with `ghRepo.fullName`

**Comment by iAmWillShepherd** [comment]
💯

**Pull request merged**